pkg/engine/engine.go

```diff
@@ -264,11 +263,8 @@ func (e *engine) invokeRuleHandler(
 				// load rule context
 				contextLoader := e.ContextLoader(policyContext.Policy(), rule)
 				if err := contextLoader(ctx, rule.Context, policyContext.JSONContext()); err != nil {
-					if _, ok := err.(gojmespath.NotFoundError); ok {
```

Comment:
Hi @fjogeleit - we want to preserve the behavior to return an error if a path is invalid (the default is to return an empty result). and 

Hence it may be necessary to maintain a fork of the JMESPath community library.

See: https://github.com/jmespath/go-jmespath/commit/b8d63dd9819bfe6e07473548bc9a37ef37a041fa

And Kyverno changes:

https://github.com/kyverno/kyverno/commit/6d0ad5598ec3905e50ee8c8d8a272fe5afd8eacb

https://github.com/kyverno/kyverno/commit/6d0ad5598ec3905e50ee8c8d8a272fe5afd8eacb

May be good to look at all other changes in our fork here for compatibility:

https://github.com/jmespath/go-jmespath/compare/master...kyverno:go-jmespath:master